docker/Dockerfile

```diff
@@ -1,64 +1,51 @@
-# 'builder' includes all build & testing dependencies.
-FROM debian:12 AS builder
+# 'builder' extends the fenv base image with FQL build & testing dependencies.
+# This stage is used by fenv for local development and CI/CD verification tasks.
+ARG FENV_DOCKER_TAG
+FROM fenv:${FENV_DOCKER_TAG} AS builder
 
-RUN apt-get update &&\
+# Install additional system packages needed for build
+RUN apt-get update && \
     apt-get install --no-install-recommends -y \
       build-essential=12.9 \
       ca-certificates=20* \
       git=1:2.39.* \
-      curl=7.88.* &&\
-    apt-get clean &&\
+      curl=7.88.* && \
+    apt-get clean && \
     rm -rf /var/lib/apt/lists/*
```

Comment:
Double check the makefile in `fenv` and ensure we aren't performing redundant installations here.